File: common/src/main/java/org/sandboxpowered/silica/network/play/serverbound/PlayerMovement.java
package org.sandboxpowered.silica.network.play.serverbound;

import org.sandboxpowered.silica.network.PacketByteBuf;
import org.sandboxpowered.silica.network.PacketHandler;
import org.sandboxpowered.silica.network.PacketPlay;
import org.sandboxpowered.silica.network.PlayConnection;

public class PlayerMovement implements PacketPlay {
    private boolean onGround;

    public PlayerMovement() {
    }

    public PlayerMovement(boolean onGround) {
        this.onGround = onGround;
    }

    @Override
    public void read(PacketByteBuf buf) {
        onGround = buf.readBoolean();
    }

    @Override
    public void write(PacketByteBuf buf) {
        buf.writeBoolean(onGround);
    }

    @Override
    public void handle(PacketHandler packetHandler, PlayConnection connection) {

    }
}

File: common/src/main/kotlin/org/sandboxpowered/silica/network/play/serverbound/PlayerMovement.kt
package org.sandboxpowered.silica.network.play.serverbound

import org.sandboxpowered.silica.network.PacketByteBuf
import org.sandboxpowered.silica.network.PacketHandler
import org.sandboxpowered.silica.network.PacketPlay
import org.sandboxpowered.silica.network.PlayContext

class PlayerMovement(private var onGround: Boolean = false) : PacketPlay {

    override fun read(buf: PacketByteBuf) {
        onGround = buf.readBoolean()
    }

    override fun write(buf: PacketByteBuf) {
        buf.writeBoolean(onGround)
    }

    override fun handle(packetHandler: PacketHandler, context: PlayContext) {}
}